core/src/main/java/org/apache/calcite/rel/metadata/RelMdColumnUniqueness.java

```diff
@@ -254,6 +255,13 @@ public Boolean areColumnsUnique(Join rel, RelMetadataQuery mq,
     final ImmutableBitSet leftColumns = leftAndRightColumns.left;
     final ImmutableBitSet rightColumns = leftAndRightColumns.right;
 
+    // for FULL OUTER JOIN if columns contain column from both inputs it is not
+    // guranteed that the result will be unique
+    if (!ignoreNulls && rel.getJoinType() == JoinRelType.FULL
+        && leftColumns.cardinality() > 0 && rightColumns.cardinality() > 0) {
+      return false;
+    }
```

Comment:
Should return null(unknown)?

Reply:
@chunweilei This API return NULL only if `areColumnsUnique` return NULL on either of the input. I attempted to kept it that way. e.g. If column is coming from left side and it is LEFT OUTER JOIN `areColumnUnique` return false.

Reply:
@vineetgarg02 , in the javadoc it is mentioned:
```
   * @return whether the columns are unique, or
   * null if not enough information is available to make that determination
```
In this specific case, you are not sure whether columns are unique or not, hence I believe you should return null.

Reply:
@jcamachor It looks like the code and comments are at odd for outer joins. Left and Right outer join return false if the column belongs to null generating side irrespective of whether columns are unique or not.

```
if (rel.getJoinType().generatesNullsOnLeft()) {
        return false;
}
```
I have tried to keep the behavior consistent for OUTER joins.

Reply:
I see, makes sense. However, it seems comments are not consistent wrt behavior. Probably there is not a rule that makes a distinction between 'false' and 'null' right now, so it is fine. Maybe you can create a follow-up (you can leave it unassigned) so we can discuss and fix it. I will merge this patch shortly.